openghg/store/base/_base.py

```diff
@@ -494,7 +523,7 @@ def _validate_datasources(
         self,
         data: list[MetadataAndData],
         fn_input_parameters: dict,
-        filepath: Path | list[Path] | None = None,
+        filepath: str | Path | list[str] | list[Path] | None = None,
```

Comment:
Do these additional types need to be added back in? With some careful prior arrangement before calling this function it would be good to ensure this is still `Path | list[Path] | None` at this point.